bertopic/plotting/_barchart.py

```diff
@@ -46,17 +48,24 @@ def visualize_barchart(topic_model,
     colors = itertools.cycle(["#D55E00", "#0072B2", "#CC79A7", "#E69F00", "#56B4E9", "#009E73", "#F0E442"])
 
     # Select topics based on top_n and topics args
-    freq_df = topic_model.get_topic_freq()
-    freq_df = freq_df.loc[freq_df.Topic != -1, :]
+    df_topic_info = topic_model.get_topic_info()
+    df_topic_info = df_topic_info.loc[df_topic_info.Topic != -1, :]
+    topic_label_map = {
+        df_topic_info.Topic.iloc[i] : df_topic_info.CustomName.iloc[i]
+        for i in range(len(df_topic_info))
+    }
     if topics is not None:
         topics = list(topics)
     elif top_n_topics is not None:
-        topics = sorted(freq_df.Topic.to_list()[:top_n_topics])
+        topics = sorted(df_topic_info.Topic.to_list()[:top_n_topics])
     else:
-        topics = sorted(freq_df.Topic.to_list()[0:6])
+        topics = sorted(df_topic_info.Topic.to_list()[0:6])
 
     # Initialize figure
-    subplot_titles = [f"Topic {topic}" for topic in topics]
+    if custom_labels:
+        subplot_titles = [f"{topic_label_map[topic]}" for topic in topics]
+    else:
+        subplot_titles = [f"Topic {topic}" for topic in topics]
```

Comment:
I think you would only need to do the following in order to use the custom labels:

```python
# Create subplot_titles 
if topic_model.custom_labels is not None and custom_labels:
    subplot_titles = [topic_model.custom_labels[topic + topic_model._outliers] for topic in topics]
else:
    subplot_titles = [f"Topic {topic}" for topic in topics]
```

So the creation of `topic_label_map` is not necessary.